File: PolicyPlus/OpenAdmxFolder.vb
Imports System.DirectoryServices.ActiveDirectory
Public Class OpenAdmxFolder
    Dim SysvolPolicyDefinitionsPath As String = ""
    Public SelectedFolder As String
    Public ClearWorkspace As Boolean
    Private Sub Options_CheckedChanged(sender As Object, e As EventArgs) Handles OptCustomFolder.CheckedChanged, OptSysvol.CheckedChanged, OptLocalFolder.CheckedChanged
        Dim customSelected = OptCustomFolder.Checked
        TextFolder.Enabled = customSelected
        ButtonBrowse.Enabled = customSelected
    End Sub
    Private Sub OpenAdmxFolder_Shown(sender As Object, e As EventArgs) Handles MyBase.Shown
        OptCustomFolder.Checked = True
        Dim compDomain As Domain = Nothing
        Try
            compDomain = Domain.GetComputerDomain
        Catch ex As Exception
            ' Not domain-joined, or no domain controller is available
        End Try
        If compDomain Is Nothing Then
            SysvolPolicyDefinitionsPath = ""
        Else
            Dim possiblePath = "\\" & compDomain.Name & "\SYSVOL\" & compDomain.Name & "\Policies\PolicyDefinitions"
            If IO.Directory.Exists(possiblePath) Then SysvolPolicyDefinitionsPath = possiblePath Else SysvolPolicyDefinitionsPath = ""
        End If
        OptSysvol.Enabled = (SysvolPolicyDefinitionsPath <> "")
    End Sub
    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        If OptLocalFolder.Checked Then
            SelectedFolder = Environment.ExpandEnvironmentVariables("%windir%\PolicyDefinitions")
        ElseIf OptSysvol.Checked Then
            SelectedFolder = SysvolPolicyDefinitionsPath
        ElseIf OptCustomFolder.Checked Then
            SelectedFolder = TextFolder.Text
        End If
        If IO.Directory.Exists(SelectedFolder) Then
            ClearWorkspace = ClearWorkspaceCheckbox.Checked
            DialogResult = DialogResult.OK
        Else
            MsgBox("The folder you specified does not exist.", MsgBoxStyle.Exclamation)
        End If
    End Sub
    Private Sub ButtonBrowse_Click(sender As Object, e As EventArgs) Handles ButtonBrowse.Click
        Using fbd As New FolderBrowserDialog
            If fbd.ShowDialog <> DialogResult.OK Then Exit Sub
            TextFolder.Text = fbd.SelectedPath
        End Using
    End Sub
    Private Sub OpenAdmxFolder_KeyUp(sender As Object, e As KeyEventArgs) Handles Me.KeyUp
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
End Class

File: PolicyPlus/OpenAdmxFolder.cs
using System;
using System.DirectoryServices.ActiveDirectory;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PolicyPlus
{
    public partial class OpenAdmxFolder
    {
        private string SysvolPolicyDefinitionsPath = "";
        public string SelectedFolder;
        public bool ClearWorkspace;

        public OpenAdmxFolder()
        {
            InitializeComponent();
        }
        private void Options_CheckedChanged(object sender, EventArgs e)
        {
            bool customSelected = OptCustomFolder.Checked;
            TextFolder.Enabled = customSelected;
            ButtonBrowse.Enabled = customSelected;
        }
        private void OpenAdmxFolder_Shown(object sender, EventArgs e)
        {
            OptCustomFolder.Checked = true;
            Domain compDomain = null;
            try
            {
                compDomain = Domain.GetComputerDomain();
            }
            catch (Exception ex)
            {
                // Not domain-joined, or no domain controller is available
            }
            if (compDomain is null)
            {
                SysvolPolicyDefinitionsPath = "";
            }
            else
            {
                string possiblePath = @"\\" + compDomain.Name + @"\SYSVOL\" + compDomain.Name + @"\Policies\PolicyDefinitions";
                if (System.IO.Directory.Exists(possiblePath))
                    SysvolPolicyDefinitionsPath = possiblePath;
                else
                    SysvolPolicyDefinitionsPath = "";
            }
            OptSysvol.Enabled = !string.IsNullOrEmpty(SysvolPolicyDefinitionsPath);
        }
        private void ButtonOK_Click(object sender, EventArgs e)
        {
            if (OptLocalFolder.Checked)
            {
                SelectedFolder = Environment.ExpandEnvironmentVariables(@"%windir%\PolicyDefinitions");
            }
            else if (OptSysvol.Checked)
            {
                SelectedFolder = SysvolPolicyDefinitionsPath;
            }
            else if (OptCustomFolder.Checked)
            {
                SelectedFolder = TextFolder.Text;
            }
            if (System.IO.Directory.Exists(SelectedFolder))
            {
                ClearWorkspace = ClearWorkspaceCheckbox.Checked;
                DialogResult = DialogResult.OK;
            }
            else
            {
                Interaction.MsgBox("The folder you specified does not exist.", MsgBoxStyle.Exclamation);
            }
        }
        private void ButtonBrowse_Click(object sender, EventArgs e)
        {
            using (var fbd = new FolderBrowserDialog())
            {
                if (fbd.ShowDialog() != DialogResult.OK)
                    return;
                TextFolder.Text = fbd.SelectedPath;
            }
        }
        private void OpenAdmxFolder_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
    }
}